plasma/plasma-mvp/morevp-alice-bob.md

```diff
@@ -0,0 +1,103 @@
+MoreVP Alice-Bob Scenarios
+---
+
+Note that these scenarios work with N participants, not just limited to 1 input 1 output. 
+
+Also, note that in all of these scenarios the exits are given the priority of the youngest input to the transaction.
+
+#### Alice & Bob are honest and cooperating:
+
+Happy path.
+
+1. Alice spends UTXO1 in TX1 to Bob.
+2. Operator withholds, neither Alice nor Bob know if the transaction has been included. 
+3. Someone with access to TX1 (Alice, Bob, or otherwise) starts an exit and places `exit bond`.
+4. Alice and Bob "piggyback" onto the exit and each put up `piggyback bond`.
+5. After 1 week, Bob receives the money as if the transaction were included. All bonds are refunded.
+
+
+#### Bob spends:
+
+Bob trying to cheat.
+
+1. Alice spends UTXO1 in TX1 to Bob, creating UTXO2.
+2. TX1 is included in block N.
+3. Bob spends UTXO2 in TX2.
+4. TX2 is included in block N+M.
+5. Bob starts an exit from TX1.
+6. Bob "piggybacks" onto the exit and places `piggyback bond`
+7. Someone reveals TX2 spending UTXO2, so Bob's `piggyback bond` is slashed.
+
+
+#### Alice double spends:
+
+This is the case where Alice is malicious and has a double-spend secretly included before her transaction to Bob.
+
+1. Alice spends UTXO1 in TX1 to Bob.
+2. Operator withholds, neither Alice nor Bob know if the transaction has been included. 
+3. Unbeknownst to Bob, Alice has double spent UTXO1 in TX2. TX2 is included in the withheld block, but TX1 isn't.
+3. Someone with access to TX1 (Alice, Bob, or otherwise) starts an exit and puts up `exit bond`.
+4. Alice and Bob "piggyback" onto the exit and each put up `piggyback bond`.
+5. Within 3.5 days, the operator reveals TX2. 
+6. No one is able to reveal TX1 in response, so Bob's `piggyback bond` is refunded and Bob doesn't get to exit. Alice's `piggyback bond` is slashed. Operator gets `exit bond` for (5). 
+
+#### Alice spends later:
+
+Here we're preventing Alice from creating a later spend that steals money from Bob.
+
+1. Alice spends UTXO1 in TX1 to Bob.
+2. TX1 is included in block N.
+3. Later, Alice double spends UTXO1 in TX2 to Carol in block N+M (chain now invalid).
+4. Alice attempts an exit on TX1 and places `exit bond`.
+5. Alice "piggybacks" onto the exit and puts up `piggyback bond`.
+6. Within 3.5 days, the someone reveals TX2, requiring that someone else reveal TX1 in response.
+7. Someone reveals TX1 and proves that TX1 was included before TX2, and gets `exit bond`. Alice's `piggyback bond` is refunded.
+
+#### Alice's second spend is non-canonical:
```

Comment:
ok so we have an alternative of this, as follows:

1. Alice and Carol spend (UTXO1, UTXO2) in TX1 to Bob
2. Carol double spends UTXO2 in TX2 to herself
3. TX1 isn't or is included, but regardless, Alice doesn't see it in blocks
4. TX2 is included, but block is withheld to Alice

If we understand correctly, Alice would in such case start an in-flight exit with TX1 and piggyback on her input, but TX1 isn't canonical (TX2 is - but that's unknown to Alice). What should Alice do, to remain unharmed?

Reply:
This is the case in which Alice cannot remained unharmed (the griefing case). She will lose the initial bond required to start the in-flight exit.

Reply:
hm, that sounds like something bigger than griefing, since the attacker gets a reward (I assume griefing = hurting the network with minor hurting of self).

Reply:
note after the call, clarified by @kfichter as follows:

Since Alice piggybacked, and that piggybacking is orthogonal to the "canonical/non-canonical" decision, this means that tx1 will be dubbed "non-canonical" and Alice's bond will be slashed. However, by the clarification, this means that Alice will exit her UTXO1 input to TX1 just fine (because TX1 non-canonical).

(note2: let's remember to adhere scrupulously to the terminology canonical/non-canonical instead of valid/invalid, that was used in this context) 

Under the assumption that the exit_bond is small - order of magnitude of the gas cost to challenge - we can say that this ends up being a "surcharge" over Alice's usual exiting costs - if Alice has a lot at stake with UTXO1, she'll follow protocol. So not as bad as I implied.

Reply:
**but** I think there's a different discrepancy here. In the other thread you said @kfichter:

>  @pdobacz
> pdobacz 8 days ago Member
> 
> how is operator incentivized (except the bond-bounty) to reveal TX2? He's not exposed to fund loss
> @kfichter
> kfichter 7 days ago Member
> 
> Whoever reveals TX2 receives exit bond, basically.

which is conflicting with the exit bond being small.

Reply:
@kfichter oh, please, take a look at this issue, in the 2 comments immediately above this one.

Reply:
The bond needs to cover the cost to challenge. This cost is gas + some additional expenses. More work needs to go into actually determining the value of the bond. I think there's also a validator's dilemma where only one validator can get the reward but we want many validators to check stuff.

Reply:
NOTE after the call: let's back track to https://github.com/omisego/research/pull/42/files#diff-8c711f0e8f03f813d5206348f119482fR32, and see if this issue can't be solved by having another, "natural" incentive around revealing TX2. Or, even better, find out this is a non-issue altogether.

Otherwise, let's be overt about the attack vector - proposition: say that medium-sized transactions that are in-flight are jeopardized, if the operator decides to withhold and is a party to a multilateral transfer (i.e. make it a known issue)